packages/cli/generators/repository/index.js

```diff
@@ -0,0 +1,393 @@
+        for (let c of b) {
+          if (c.includes('type')) {
+            const tempVal = c.replace('type:', '').trim();
+            // let's make sure we parsed the datatype correctly
+            if (VALID_PROPERTY_ID_TYPES.includes(tempVal)) {
+              retVal = tempVal;
+            }
+            break;
+          }
+        }
+      }
+    }
+
+    return retVal;
+  }
+
+  /**
+   * load the connectors available and check if the basedModel matches any
+   * connectorType supplied for the given connector name
+   * @param {string} connectorType single or a comma separated string array
+   */
+  async _isConnectorOfType(connectorType, dataSourceClassName) {
+    debug(`calling isConnectorType ${connectorType}`);
+    let jsonFileContent = '';
+    let result = false;
+
+    let datasourceJSONFile = path.join(
+      this.artifactInfo.datasourcesDir,
+      dataSourceClassName
+        .replace('Datasource', '.datasource.json')
+        .toLowerCase(),
+    );
+
+    try {
+      jsonFileContent = this.fs.readJSON(datasourceJSONFile, {});
+    } catch (err) {
+      debug(`${ERROR_READING_FILE} ${datasourceJSONFile}: ${err.message}`);
+      return this.exit(err);
+    }
+
+    for (let connector of Object.values(connectors)) {
+      if (
+        jsonFileContent.connector === connector.name &&
+        connectorType.includes(connector.baseModel)
+      ) {
+        result = true;
+        break;
+      }
+    }
+
+    return result;
+  }
+
+  _setupGenerator() {
+    super._setupGenerator();
+
+    this.artifactInfo = {
+      type: 'repository',
+      rootDir: 'src',
+    };
+    this.artifactInfo.outDir = path.resolve(
+      this.artifactInfo.rootDir,
+      'repositories',
+    );
+    this.artifactInfo.datasourcesDir = path.resolve(
+      this.artifactInfo.rootDir,
+      'datasources',
+    );
+    this.artifactInfo.modelDir = path.resolve(
+      this.artifactInfo.rootDir,
+      'models',
+    );
+    this.artifactInfo.defaultTemplate = REPOSITORY_CRUD_TEMPLATE;
+  }
+
+  setOptions() {
+    return super.setOptions();
+  }
+
+  checkLoopBackProject() {
+    return super.checkLoopBackProject();
+  }
+
+  async promptDataSourceName() {
+    debug('Prompting for a datasource ');
+    let datasourcesList;
+
+    try {
+      datasourcesList = await utils.getArtifactList(
+        this.artifactInfo.datasourcesDir,
+        'datasource',
+        true,
+      );
+    } catch (err) {
+      return this.exit(err);
+    }
+
+    // iterate over it to exclude service oriented data sources
+    let tempDataSourceList = Object.assign(datasourcesList, {});
+    for (let item of tempDataSourceList) {
+      let result = await this._isConnectorOfType(
+        VALID_CONNECTORS_FOR_REPOSITORY,
+        item,
+      );
+      debug(
+        `${item} has keyword ${VALID_CONNECTORS_FOR_REPOSITORY} is ${result}`,
+      );
+      if (!result) {
+        // remove from original list
+        _.remove(datasourcesList, e => e == item);
```

Comment:
Please rewrite this look using `Array.prototype.filter` function.

```ts
const availableDatasources = datasourcesList.filter(item => {
  const result = await this._isConnectorOfType(
     VALID_CONNECTORS_FOR_REPOSITORY,
     item,
   );
   debug(/*...*/);
   return result;
});
```

Reply:
@bajtos , sure, I will update it, but what would be the benefit ? .

Reply:
The code is easier to reason about because you are not modifying the array items in the middle of the loop iterating through it, and you also don't need `_.remove` from an external library.